framework/db/conditions/LikeConditionBuilder.php

```diff
@@ -78,7 +78,7 @@ public function build(ExpressionInterface $expression, array &$params = [])
             if ($value instanceof ExpressionInterface) {
                 $phName = $this->queryBuilder->buildExpression($value, $params);
             } else {
-                $phName = $this->queryBuilder->bindParam(empty($escape) ? $value : ('%' . strtr($value, $escape) . '%'), $params);
+                $phName = $this->queryBuilder->bindParam(empty($escape) || empty($value) ? $value : ('%' . strtr($value, $escape) . '%'), $params);
```

Comment:
It has the same problem as mentioned in https://github.com/yiisoft/yii2/pull/19685#discussion_r1027371082 - if you search `'0'`, it suddenly won't work as expected. 

Also, still nobody explained how query should look like if you pass `null` here - https://github.com/yiisoft/yii2/pull/19685#discussion_r1061298539

Reply:
Yes, null comparison should be used here. And we don't want `like '%%'`.

Reply:
@rob006 @bizley I replaced `empty()` with `is_null()`

Reply:
> Yes, null comparison should be used here. And we don't want `like '%%'`.

But `LIKE ""` does not make sense either.

Reply:
I suppose that's another issue which existed before

Reply:
Previously it generated `LIKE '%%'`, which in practice matches everything except `null`. Now it will generate `LIKE ''`, which matches only empty string - this is not only a BC break, but IMO it is even more nonsensical than `LIKE '%%'`

Reply:
So maybe let's check for LIKE operator and null/empty string as well.

Reply:
@rob006 @bizley I made empty $value as a separate case